drivers/rtc/rtc_rpi_pico.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/irq.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <hardware/rtc.h>
+
+#define DT_DRV_COMPAT raspberrypi_pico_rtc
+
+/* struct tm start time:   1st, Jan, 1900 */
+#define TM_YEAR_REF 1900
+
+struct rtc_rpi_pico_data {
+	struct k_mutex lock;
+};
+
+LOG_MODULE_REGISTER(rtc_rpi, CONFIG_RTC_LOG_LEVEL);
+
+static int rtc_rpi_pico_init(const struct device *dev)
+{
+	struct rtc_rpi_pico_data *data = dev->data;
+
+	k_mutex_init(&data->lock);
```

Comment:
You should use a k_spinlock instead of a mutex since you don't need to relock or track ownership of the lock :)